src/pages/iv-materials-tfr/Form/MaterialsTransferForm.js

```diff
@@ -0,0 +1,913 @@
+    enabled: !recordId
+  })
+
+  const initialValues = {
+    recordId: null,
+    functionId: SystemFunction.MaterialTransfer,
+    reference: '',
+    dtId: documentType?.dtId,
+    oDocId: '',
+    date: new Date(),
+    closedDate: null,
+    receivedDate: null,
+    fromSiteId: '',
+    toSiteId: '',
+    notes: '',
+    status: 1,
+    plantId: parseInt(plantId),
+    printStatus: null,
+    qty: null,
+    pcs: null,
+    isVerified: false,
+    wip: 1,
+    carrierId: null,
+    transfers: [
+      {
+        id: 1,
+        transferId: recordId || 0,
+        seqNo: 1,
+        sku: '',
+        itemName: null,
+        itemId: null,
+        qty: null,
+        componentSeqNo: 0,
+        qtyInBase: '',
+        muId: null,
+        muQty: '',
+        muRef: '',
+        weight: null,
+        unitCost: 0,
+        msId: null,
+        trackBy: null,
+        lotCategoryId: null,
+        metalId: null,
+        totalCost: 0,
+        priceType: null
+      }
+    ]
+  }
+
+  const invalidate = useInvalidate({
+    endpointId: InventoryRepository.MaterialsTransfer.qry
+  })
+
+  const { formik } = useForm({
+    initialValues,
+    maxAccess,
+    enableReinitialize: false,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      date: yup.date().required(),
+      fromSiteId: yup.string().required(),
+      toSiteId: yup.string().required(),
+      transfers: yup
+        .array()
+        .of(
+          yup.object().shape({
+            sku: yup.string().required(),
+            qty: yup.string().required(),
+            msId: yup.string().required()
+          })
+        )
+        .required()
+    }),
+    onSubmit: async values => {
+      try {
+        const copy = { ...values }
+        delete copy.transfers
+        copy.date = !!copy.date ? formatDateToApi(copy.date) : null
+        copy.closedDate = !!copy.closedDate ? formatDateToApi(copy.closedDate) : null
+        copy.receivedDate = !!copy.receivedDate ? formatDateToApi(copy.receivedDate) : null
+        copy.status = copy.status === '' ? 1 : copy.status
+        copy.wip = copy.wip === '' ? 1 : copy.wip
+
+        const updatedRows = formik?.values?.transfers.map((transferDetail, index) => {
+          return {
+            ...transferDetail,
+            seqNo: index + 1,
+            transferId: formik.values.recordId || 0
+          }
+        })
+        if (values.fromSiteId === values.toSiteId) {
+          stackError({
+            message: `Cannot have same from and to site`
+          })
+
+          return
+        }
+
+        const resultObject = {
+          header: copy,
+          items: updatedRows,
+          serials: [],
+          lots: []
+        }
+
+        let notificationData
+
+        const res = await postRequest({
+          extension: InventoryRepository.MaterialsTransfer.set2,
+          record: JSON.stringify(resultObject)
+        })
+
+        if (!values.recordId) {
+          toast.success(platformLabels.Added)
```

Comment:
please after set either new mode or edit mode refill the whole form not only recordId and reference

Reply:
res is only returning recordId in obj